src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,662 @@
+from DIRAC.Core.Security.ProxyInfo import getVOfromProxyGroup
+from DIRAC.Resources.Computing.ARCComputingElement import ARCComputingElement
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+from DIRAC.Core.Security.X509Chain import X509Chain  # pylint: disable=import-error
+from DIRAC.Core.Security import Locations
+from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
+from DIRAC.ConfigurationSystem.Client.Helpers.Path import cfgPath
+
+CE_NAME = "AREX"
+MANDATORY_PARAMETERS = ["Queue"]  # Mandatory for ARC CEs
+
+# Note : interiting from ARCComputingElement. See https://github.com/DIRACGrid/DIRAC/pull/5330#discussion_r740907255
+class AREXComputingElement(ARCComputingElement):
+
+    # Used in getJobStatus
+    mapStates = {
+        "Accepted": "Scheduled",
+        "Preparing": "Scheduled",
+        "Submitting": "Scheduled",
+        "Queuing": "Scheduled",
+        "Undefined": "Unknown",
+        "Running": "Running",
+        "Finishing": "Running",
+        "Deleted": "Killed",
+        "Killed": "Killed",
+        "Failed": "Failed",
+        "Hold": "Failed",
+        "Finished": "Done",
+        "Done": "Done",
+        "Other": "Done",
+    }
+
+    #############################################################################
+    def __init__(self, ceUniqueID):
+        """Standard constructor."""
+        super(AREXComputingElement, self).__init__(ceUniqueID)
+
+        self.ceType = CE_NAME
+        self.submittedJobs = 0
+        self.mandatoryParameters = MANDATORY_PARAMETERS
+        self.pilotProxy = ""
+        self.queue = ""
+        self.outputURL = "gsiftp://localhost"
+        self.ceHost = self.ceName
+
+        result = returnValueOrRaise(getCESiteMapping(self.ceHost))
+        self.site = result[self.ceHost]
+        self.grid = self.site.split(".")[0]
+
+        # For the REST interface
+        if self.initialised:  # a bunch of the local variables are set up here.
+            self.log.info("REST interface successfully configured", "for %s" % self.ceHost)
+        else:
+            self.log.info("REST interface not yet properly configured", "for %s" % self.ceHost)
+
+        # Do these after all other initialisations, in case something barks
+        self.xrslExtraString = self.__getXRSLExtraString()
+        self.xrslMPExtraString = self.__getXRSLExtraString(multiprocessor=True)
+
+        # Get from the CS the time left (should be in seconds) for proxy before renewal
+        self.proxyTimeLeftBeforeRenewal = 10000  # 2 hours, 46 minutes and 40 seconds
+        variable = "proxyTimeLeftBeforeRenewal"
+        firstOption = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, variable)
+        secondOption = cfgPath("Resources", "Sites", self.grid, self.site, variable)
+        defaultOption = cfgPath("Resources", "Computing" "CEDefaults", variable)
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
```

Comment:
Shouldn't this process of finding options already be done for you? The `self.ceParameters` dictionary should have them already, IIUC.

Reply:
Thanks - to be checked by me and updated / fixed.

It is the same algorithm also used in the ARCComputingElement.

Reply:
Done.